projects/packages/forms/src/dashboard/inbox/dataviews/index.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * External dependencies
+ */
+import {
+	ExternalLink,
+	Modal,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { useResizeObserver } from '@wordpress/compose';
+import { useEntityRecords } from '@wordpress/core-data';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { DataViews } from '@wordpress/dataviews/wp';
+import { dateI18n } from '@wordpress/date';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+/**
+ * Internal dependencies
+ */
+import { store as dashboardStore } from '../../store';
+import InboxResponse from '../response';
+import {
+	viewAction,
+	markAsSpamAction,
+	markAsNotSpamAction,
+	moveToTrashAction,
+	deleteAction,
+	restoreAction,
+} from './actions';
+import { useView, defaultLayouts } from './views';
+
+const EMPTY_ARRAY = [];
+const EMPTY_OBJECT = {};
+const MOBILE_BREAKPOINT = 780;
+const getItemId = item => item.id.toString();
+const getPath = response => {
+	try {
+		const url = new URL( response.entry_permalink );
+		return url.pathname;
+	} catch {
+		return '';
+	}
+};
```

Comment:
Nit: probably deserves a doc block, just like the function that follows it. The argument name `response` may also be a distraction, since the domain-specific term is usually `item`.

Reply:
I updated it, but in general I just copied this function here. It was in a `utils` file and it wasn't reused anywhere else.